lib/matplotlib/cbook.py

```diff
@@ -2406,6 +2407,132 @@ def get_instancemethod(self):
+
+    is the same as ::
+
+       wrapped = mappable_kwargs_decorator(func, {'x': 'foo', 'y':'bar'})
+       wrapped(ax, data)
+
+    Parameters
+    ----------
+    function : callable
+
+        A function which expects an `Axes` objects as the first argument
+        and can take data as kwargs.
+
+    map_targets : dict
+
+       Mapping between key expected by the function and the column name.
+       ex ``{'x': 'foo', 'y': 'bar'}`` ->
+       ``kwargs['x'] == data['foo']`` and ``kwargs['y'] == data['bar']``
+
+
+    Returns
+    -------
+    callable
+
+       Decorated function with the signature ::
+
+          func(ax, df, *args, **kwargs)
+
+
+    """
+    @wraps(func)
+    def inner(ax, data, *args, **kwargs):
+        for k, v in map_targets.items():
+            if k in kwargs:
+                raise ValueError(("Trying to map a column ({1} -> {0}) "
+                                  "on to a passed in "
+                                  "keyword arg ({0})").format(k, v))
+            kwargs[k] = data[v].values
+        return func(ax, *args, **kwargs)
+
+    return inner
+
+
+def mappable_args_decorator(func, map_targets):
+    """Decompose labeled data for passing to base functions.
+
+    This is a decorator to easy the burden of unpacking
+    pandas data frames into base objects.
+
+    In the returned function args and kwargs are passed through and
+    the specified columns are re-mapped to precede the args.
+
+    The args and kwargs are then unpacked into the function (which will
+    expect an `Axes` as the first input and a `DataFrame`-like objects
+    as the second.
+
+
+    The function call ::
+
+        func(ax, data['foo'], data['bar'])
+
+    is the same as ::
+
+        wrapped = mappable_args_decorator(func, ('foo', 'bar'))
+        wrapped(ax, data)
+
+
+    Parameters
+    ----------
+    function : callable
+        A function which expects an `Axes` objects as the first argument
+        and can take data as kwargs.
+
+    map_targets : tuple
+       Order of columns to extract to pass to the function.  These column
+       are the 2nd -> n+1 args
+
+
+    Returns
+    -------
+    callable
+        Decorated function with the signature ::
+
+           func(ax, df, *args, **kwargs)
+
+
+    """
+    @wraps(func)
+    def inner(ax, data, *args, **kwargs):
+        args = tuple(data[k] for k in map_targets) + args
+        return func(ax, *args, **kwargs)
+
+    return inner
+
+
+def apply_kwargs_mapping(ax, func, data, map_targets, *args, **kwargs):
+    for k, v in map_targets.items():
+        if k in kwargs:
+            raise ValueError(("Trying to map a column ({1} -> {0}) "
+                              "on to a passed in "
+                              "keyword arg ({0})").format(k, v))
+        kwargs[k] = data[v].values
+        return func(ax, *args, **kwargs)
+
+
+def apply_args_mapping(ax, func, data, map_targets, *args, **kwargs):
+    args = tuple(data[k] for k in map_targets) + args
```

Comment:
Omitting `.values` also means it will work with a numpy structured array; but I'm not sure if there would be any point in this.
Maybe use `np.asanyarray` in case something might yield a masked array?
I presume with a DataFrame, missing data will end up as NaN after conversion, correct?